### PR TITLE
frontend: Implement Material 3 Theming & Tips Section

### LINKED ISSUE #6: #5 : Integrate Basic Styling and Theme
## Description
- Establish a consistent color scheme and font style for the UI.
- Ensure the design is visually appealing and accessible.

### PR DESCRIPTION
**Pull request by criticic**
fixes #6 

image
image
image
image

### CODE REVIEW

**Comment by Sweetdevil144** [comment]
Awesome. TBH I hate frontend work and I prefer backend more. That's the reason we decided to proceed with Frontend first and later on integrate with Backend.

File: client/src/App.test.tsx
Comment (Sweetdevil144) [inline client/src/App.test.tsx:1]: Can you restore this file? We'll use it later on to test out our APIs from the Front-End.

File: client/src/MainLayout.tsx [hunk 1]
```diff
@@ -1,59 +1,47 @@
...
-                   onClick={handleSend}>
-                    REQUEST
-                </button>
-            </footer>
-        </div>
-    );
+  return (
+    <div className="flex flex-col h-screen bg-surface">
+      <Header />
+      <main className="flex-1 p-4 overflow-auto">
```
Comment (Sweetdevil144) [inline client/src/MainLayout.tsx:39]: Shouldn't be `overflow-auto`! Can you correct this? The size of Input field should increase in height when the test surpasses line length. UPT a max of 3 lines. Than, you should apply `overflow-y` property

File: client/src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const Header: React.FC = () => {
+  return (
+    <header className="px-2 py-4 text-start">
+      <h1 className="text-4xl font-mono font-semibold">Cat-GPT</h1>
```
Comment (Sweetdevil144) [inline client/src/components/Header.tsx:6]: Can you add a button here too to switch from Dark to Light

File: client/src/components/Tips.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      <div className="text-start my-2">{text}</div>
+    </div>
+  );
+};
+
+const Tips: React.FC = () => {
+  const tips: TipProps[] = [
+    {
+      icon: (
+        <CodeBracketIcon className="h-6 w-6 p-1 rounded-full bg-primary-container" />
```
Comment (Sweetdevil144) [inline client/src/components/Tips.tsx:26]: Looks good to me. I have some ideas which I'll implement later on in this file. But this is good.

**Review by Sweetdevil144 — CHANGES_REQUESTED** [review]
I've left some changes to be done. Can you do that and once it's done I'll merge this.

File: client/src/App.test.tsx
Comment (criticic) [reply client/src/App.test.tsx:1]: Added it back

File: client/src/components/Header.tsx (hunk 2 above)
Comment (criticic) [reply client/src/components/Header.tsx]: done

**Comment by criticic** [comment]
image

**Review by Sweetdevil144 — APPROVED** [review]
LGTM !!

**Pull request merged**